src/mongoc/mongoc-cursor.c

```diff
@@ -239,9 +242,8 @@ _mongoc_cursor_new_with_opts (mongoc_client_t *client,
          GOTO (finish);
       }
 
+      bson_destroy (&cursor->filter);
```

Comment:
This is the same approach I saw used elsewhere when SDAM regularly updates the last `isMaster` response. Unlike `opts`, which was previously initialized for all but the error case (a straightforward patch IMO), with `filter` we're potentially initializing an inline, static document and overwriting it if the user passed a filter. This is a bit redundant, but I felt the approach was much cleaner than a conditional `bson_init()` if `filter` failed validation.